examples/counter/test/sagas.js

```diff
@@ -25,5 +26,20 @@ test('incrementAsync Saga test', (t) => {
     'counter Saga must be done'
   )
 
+  const watcher = rootSaga()
+  const worker = takeEvery('INCREMENT_ASYNC', incrementAsync);
+
+  t.deepEqual(
+    watcher.next().value,
+    worker.next().value,
+    'rootSaga takes from every INCREMENT_ASYNC action'
+  )
+
+  t.equal(
+    watcher.return().done,
```

Comment:
Hm, you are right, but there is a caveat. Look at what is returned by `takeEvery`:
```
{ name: 'takeEvery(INCREMENT_ASYNC, incrementAsync)',
      next: [Function: next],
      throw: [Function],
      return: [Function: kReturn],
      '@@redux-saga/HELPER': true }
```

And now because of this - ` '@@redux-saga/HELPER': true` its gonna be interpreted by the `redux-saga` in the special way:
`: is.helper(effect)                                    ? runForkEffect(wrapHelper(effect), effectId, currCb)`

So effectively in real saga the task descriptor is going to be injected into the saga, and not an iterator.